Pyproject.toml (#338)
simpeg/discretize: docs/content/installing.rst

is our recommended way of installing `discretize`::

    conda install -c conda-forge discretize

`discretize` is also available on pypi::

    pip install discretize

There are currently pre-built wheels for windows available on pypi, but other operating
systems will require a build.

Installing from Source
----------------------
If you are not a developer then conda-forge is really the preferred way of installing.

As `discretize` contains several compiled extensions and is not a pure python pacakge,
installing `discretize` from the source code requires a C/C++ compiler capable of
using a C++ 17 standard.

`discretize` uses a `pyproject.toml` file to define the build and install steps. As such
there is no `setup.py` file to run. You must use `pip` to install `discretize`. As long as
you have an available compiler you should be able to install `discretize` from the source as::

    pip install .

Editable Installs
^^^^^^^^^^^^^^^^^
If you are an active developer of `discretize`, and find yourself modifying the code often,
you might want to install if from source, in an editable installation. `discretize` uses

> `meson-python` to build the external modules and install the package. As such, there are several
> steps to take, first you must have the `meson-python` requirements in your environment::
> 
>     pip install meson-python meson ninja

Reviewer: Just merging the installation of build dependencies in a single line, and adding instructions for conda.

```suggestion
`meson-python` to build the external modules and install the package. As such, there are a few
steps to take. First, make sure you have the runtime dependencies installed in your environment (see Dependencies listed above). Then you must install some packages needed to build ``discretize`` in your environment. You can do so with ``pip``::

    pip install meson-python meson ninja cython setuptools_scm

Or with ``conda`` (or ``mamba``)::

    conda -c conda-forge install meson-python meson ninja cython setuptools_scm
```

This will allow you to use the build backend required by `discretize`.

> `discretize` also has its own build requirements (listed above) which must be installed,
> along with the runtime requirements::

Reviewer: Remove these lines if we accept the change from above.
```suggestion
```

>     pip install numpy cython setuptools_scm scipy

Reviewer: Same with this line.
```suggestion
```
